### DISCUSSION TITLE
I don't understand the firmware on my tube_zb_gw_cc2652p2_poe

### DISCUSSION THREAD

**Discussion by CharlesGillanders** [tube0013/tube_gateways#102] [Q&A]
Hi, I have a tube_zb_gw_cc2652p2_poe - within the ESPHome component of HomeAssistant it says the firmware on the device is
Device info
Espressif ESP-WROVER-KIT
by espressif
Firmware: 1.17.1 (May 12 2021, 22:11:35)

From what I can tell ESPHome looks like it started using a new naming convention of Year.Month.Release in August of 2021 - am I able to update the ESPHome firmware on my system?  Should I do so?  I was able to update the zigbee firmware using cc2538-bsl.py but the ESPHome firmware is confusing me.

**Comment by tube0013** [comment]
ESPhome releases after 2021.9 made the custom serial stream component that is used and relied on unstable prone to disconnects often. I believe the versions even earlier than this were more stable.

I recently contracted out a developer to fix this. and have a new version that works with current ESPhome available. It's still marked as beta but in my testing I have found it to be stable.

if you go to the main page: https://github.com/tube0013/tube_gateways

and click on your device mode, you will see a folder for Firmware and within the esphome folder is a beta folder with the new yaml if you desire to compile yourself or a binary.

**Reply by CharlesGillanders** [reply]
Thanks - I can certainly look at compiling that.

Practically to get the controller updated with that complied code do I use the OTA update function on the built in web server or do I need to follow the instructions to serially flash the ESP32 module?

**Reply by tube0013** [reply]
you can use the OTA flash on the built in web server. the serial flash is useful for doing a full erase of the flash on the esp32 - there are cases where this can solve issues.
